File: Easy/Skynet the Chasm/Python/Skynet the Chasm.py
import sys

lenght_of_road        = int(raw_input())
lenght_of_gap         = int(raw_input())  
lenght_of_platform    = int(raw_input())  

while True:
    motorbike_speed     = int(raw_input())  
    motorbike_position  = int(raw_input()) 
    
    if motorbike_position > lenght_of_road or motorbike_speed > lenght_of_gap + 1:
        answer = "SLOW"
    elif motorbike_speed < lenght_of_gap + 1:
        answer = "SPEED"
    elif motorbike_position == lenght_of_road - 1:
        answer = "JUMP"
    else:
        answer = "WAIT"
    
    print(answer)

File: Easy/Skynet the Chasm/Ruby/SkynetTheChasm.rb
STDOUT.sync = true

@lenght_of_road     = gets.to_i
@lenght_of_gap      = gets.to_i
@lenght_of_platform = gets.to_i

loop do
  @motorbike_speed    = gets.to_i
  @motorbike_position = gets.to_i

  if (@motorbike_position > @lenght_of_road or @motorbike_speed > @lenght_of_gap + 1)
    puts "SLOW"
  elsif (@motorbike_speed < @lenght_of_gap + 1)
    puts "SPEED"
  elsif (@motorbike_position == @lenght_of_road - 1)
    puts "JUMP"
  else
    puts "WAIT"
  end
end